SnM/SnM_Notes.cpp

```diff
@@ -1352,17 +1360,184 @@ bool ImportSubRipFile(const char* _fn)
 	return ok;
 }
 
+// see http://moodub.free.fr/video/ass-specs.doc
+bool ImportAdvancedSubStationFile(const char* _fn)
+{
+    bool ok = false;
+    double firstPos = -1.0;
+
+    // no need to check extension here, it's done for us
+    if (FILE* f = fopenUTF8(_fn, "rt"))
+    {
+        char bom[4];
+        if (!fgets(bom, 4, f) || !*bom) {
+            fclose(f);
+            return ok;
+        }
+        if (!(bom[0] == char(0xEF) && bom[1] == char(0xBB) && bom[2] == char(0xBF))) {
+            rewind(f);
+        }
+        char buf[1024];
+        while(fgets(buf, sizeof(buf), f) && *buf) // searching for Events block
+        {
+            if (strncmp(buf, "[Events]", 8) == 0) // searching for events block, we don't care about styles and stuff
+            {
+                break;
+            }
+        }
+        if (!(fgets(buf, sizeof(buf), f) && *buf))
+            return false; // reading and discarding format line, we're only interested in fields that are fixed
+
+        while(fgets(buf, sizeof(buf), f) && *buf) // reading events
+        {
+            int p1[2], p2[2];
+            float sec1, sec2;
+            int i = 0, j = 0;
+            string s;
+            char tokens[10][64];
+            while (buf[i] != '\0' && j < 9) { // reading event fields in this dumb way because sscanf doesn't work
+                if (buf[i] != ',')  // basically like python's buf.split(',', maxsplit=9)
+                {
+                    s += buf[i];
+                }
+                else
+                {
+                    strcpy(tokens[j], s.c_str());
```

Comment:
I'm not familiar with the ASS format so I don't know how big it needs to be to not lose data. The important part though is not its size but that it must be impossible to write beyond its end.

Perhaps this could be done differently? Only the second, third and fifth tokens are used so there's no need to store each of them in a separate string. Eg. `sccanf(buf + offset_to_first_comma)`.